src/generators/template/TemplateGenerator.ts

```diff
@@ -0,0 +1,116 @@
+      this.options.typeMapping, 
+      this.options.constraints, 
+      {
+        metaModel: model,
+        options: this.options,
+        constrainedName: '' //This is just a placeholder, it will be constrained within the function
+      }
+    );
+  }
+
+  /**
+   * Render a scattered model, where the source code and library and model dependencies are separated.
+   * 
+   * @param model 
+   * @param inputModel 
+   */
+  render(model: ConstrainedMetaModel, inputModel: InputMetaModel): Promise<RenderOutput> {
+    if (model instanceof ConstrainedObjectModel) {
+      return this.renderClass(model, inputModel);
+    } else if (model instanceof ConstrainedEnumModel) {
+      return this.renderEnum(model, inputModel);
+    } 
+    Logger.warn(`Template generator, cannot generate this type of model, ${model.name}`);
+    return Promise.resolve(RenderOutput.toRenderOutput({ result: '', renderedName: '', dependencies: [] }));
+  }
+
+  /**
+   * Render a complete model result where the model code, library and model dependencies are all bundled appropriately.
+   * 
+   * For Template you need to specify which package the model is placed under.
+   * 
+   * @param model 
+   * @param inputModel 
+   * @param options used to render the full output
+   */
+  async renderCompleteModel(model: ConstrainedMetaModel, inputModel: InputMetaModel, options: TemplateRenderCompleteModelOptions): Promise<RenderOutput> {
+    if (isReservedTemplateKeyword(options.packageName)) {
+      throw new Error(`You cannot use reserved Template keyword (${options.packageName}) as package name, please use another.`);
+    }
+
+    const outputModel = await this.render(model, inputModel);
+    const modelDependencies = model.getNearestDependencies().map((dependencyModel) => {
+      return `import ${options.packageName}.${dependencyModel.name};`;
+    });
+    const outputContent = `package ${options.packageName};
+${modelDependencies.join('\n')}
+${outputModel.dependencies.join('\n')}
+${outputModel.result}`; 
+    return RenderOutput.toRenderOutput({result: outputContent, renderedName: outputModel.renderedName, dependencies: outputModel.dependencies});
+  }
+
+  async renderClass(model: ConstrainedObjectModel, inputModel: InputMetaModel): Promise<RenderOutput> {
+    const presets = this.getPresets('class');
+    const renderer = new ClassRenderer(this.options, this, presets, model, inputModel);
+    const result = await renderer.runSelfPreset();
+    return RenderOutput.toRenderOutput({result, renderedName: model.name, dependencies: renderer.dependencies});
+  }
+
+  async renderEnum(model: ConstrainedEnumModel, inputModel: InputMetaModel): Promise<RenderOutput> {
+    const presets = this.getPresets('enum'); 
+    const renderer = new EnumRenderer(this.options, this, presets, model, inputModel);
+    const result = await renderer.runSelfPreset();
+    return RenderOutput.toRenderOutput({result, renderedName: model.name, dependencies: renderer.dependencies});
+  }
```

Comment:
Hmm, one renderer type like class is enough for me. Not every language has something like enum but almost every has class or struct.

Reply:
It is not as much for the enum's sake, but to have two renderers, whether it's for the enum or something else does not really matter 🙂 So I would say it's best just to keep it, once it has been used a bit we can always bring up the discussion whether it should be removed.